### PR TITLE
OBSDOCS-939/TRACING-4067: Write documentation for kubeletstats rcv component

### PR DESCRIPTION
**Pull request by max-cx**
:warning: This PR is tied to a product release date, which is currently set to **June 5** (Wednesday).

- If the merge review gets delayed, you're welcome to ping `mleonov` on Slack.
- If you have any questions, ping `mleonov` on Slack.

This PR edits https://github.com/openshift/openshift-docs/pull/73626/

Version(s):

4.12, 4.13, 4.14, 4.15, 4.16

Issue:

https://issues.redhat.com/browse/OBSDOCS-939
https://issues.redhat.com//browse/TRACING-4067

Link to docs preview:

https://75202--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/otel/otel-configuration-of-otel-collector.html#kubeletstats-receiver_otel-configuration-of-otel-collector

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/otel-collector-components.adoc [hunk 1]
```diff
@@ -82,6 +82,55 @@ The Jaeger receiver ingests traces in the Jaeger formats.
 <4> The Jaeger Thrift Binary endpoint. If omitted, the default `+0.0.0.0:6832+` is used.
 <5> The  server-side TLS configuration. See the OTLP receiver configuration section for more details.
 
+[id="kubeletstats-receiver_{context}"]
+=== Kubelet Stats Receiver
+
+The Kubelet Stats receiver is currently a link:https://access.redhat.com/support/offerings/techpreview[Technology Preview] feature only.
+
+The Kubelet Stats receiver extracts metrics related to nodes, pods, containers, and volumes from the kubelet's API server. These metrics are then channeled through the metrics-processing pipeline for additional analysis.
```
[the remark this answers is not in the document]
Comment (max-cx) [reply modules/otel-collector-components.adoc:90]: Component name, not generic.

File: modules/otel-collector-components.adoc [hunk 2]
```diff
@@ -82,6 +82,55 @@ The Jaeger receiver ingests traces in the Jaeger formats.
...
+
+.OpenTelemetry Collector custom resource with an enabled Kubelet Stats receiver
```
[the remark this answers is not in the document]
Comment (max-cx) [reply modules/otel-collector-components.adoc:92]: Component name, not generic.

File: modules/otel-collector-components.adoc [hunk 3]
```diff
@@ -82,6 +82,55 @@ The Jaeger receiver ingests traces in the Jaeger formats.
...
+	receivers: [kubeletstats]
+env:
+  - name: K8S_NODE_NAME # <1>
+    valueFrom:
+      fieldRef:
+        fieldPath: spec.nodeName
+----
+<1> Sets the `K8S_NODE_NAME` to authenticate to the API.
+
+The Kubelet Stats receiver requires additional permissions for the service account used for running the OpenTelemetry Collector.
```
[the remark this answers is not in the document]
Comment (max-cx) [reply modules/otel-collector-components.adoc:114]: Component name, not generic.

File: modules/otel-collector-components.adoc [hunk 4]
```diff
@@ -82,6 +82,55 @@ The Jaeger receiver ingests traces in the Jaeger formats.
 <4> The Jaeger Thrift Binary endpoint. If omitted, the default `+0.0.0.0:6832+` is used.
 <5> The  server-side TLS configuration. See the OTLP receiver configuration section for more details.
 
+[id="kubeletstats-receiver_{context}"]
+=== Kubelet Stats Receiver
+
+The Kubelet Stats receiver is currently a link:https://access.redhat.com/support/offerings/techpreview[Technology Preview] feature only.
```
[the remark this answers is not in the document]
Comment (max-cx) [reply modules/otel-collector-components.adoc:88]: Component name, not generic.

File: modules/otel-collector-components.adoc [hunk 5]
```diff
@@ -82,6 +82,55 @@ The Jaeger receiver ingests traces in the Jaeger formats.
...
+rules:
+  - apiGroups: [""]
+    resources: ["nodes/stats"] # <1>
+    verbs: ["get"]
+  - apiGroups: [""]
+    resources: ["nodes/proxy"] # <2>
+    verbs: ["get"]
+----
+<1> Requires the `get` permissions.
+<2> Requires permissions when using the `extra_metadata_labels` or the `request_utilization` or `limit_utilization` metrics.
```
Comment (max-cx) [inline modules/otel-collector-components.adoc:132]: @iblancasa, about callout 2, `Requires` which `permissions`?

File: modules/otel-collector-components.adoc [hunk 6]
```diff
@@ -82,6 +82,55 @@ The Jaeger receiver ingests traces in the Jaeger formats.
...
+rules:
+  - apiGroups: [""]
+    resources: ["nodes/stats"] # <1>
+    verbs: ["get"]
+  - apiGroups: [""]
+    resources: ["nodes/proxy"] # <2>
+    verbs: ["get"]
+----
+<1> Requires the `get` permissions.
+<2> Requires additional permissions when using the `extra_metadata_labels` or the `request_utilization` or `limit_utilization` metrics.
```
Comment (max-cx) [inline modules/otel-collector-components.adoc:132]: @iblancasa, about callout 2, `Requires` which additional `permissions`? Do we need to specify which?

Comment (iblancasa) [reply modules/otel-collector-components.adoc:132]: The permissions are:
```yaml
  - apiGroups: [""]
    resources: ["nodes/proxy"] # <2>
    verbs: ["get"]
```

If you are not using what is described in `<2>`, you don't need permissions over `nodes/proxy`.

Comment (iblancasa) [reply modules/otel-collector-components.adoc:132]: So, for me it's fine.

**Review by iblancasa — APPROVED** [review]

File: modules/otel-collector-components.adoc (hunk 6 above)
Comment (max-cx) [reply modules/otel-collector-components.adoc:132]: Thank you, Israel.

Comment (max-cx) [reply modules/otel-collector-components.adoc]: I rewrote this callout.

**Review by mletalie — COMMENTED** [review]
/label peer-review-in-progress

File: modules/otel-collector-components.adoc [hunk 7]
```diff
@@ -82,6 +82,55 @@ The Jaeger receiver ingests traces in the Jaeger formats.
 <4> The Jaeger Thrift Binary endpoint. If omitted, the default `+0.0.0.0:6832+` is used.
 <5> The  server-side TLS configuration. See the OTLP receiver configuration section for more details.
 
+[id="kubeletstats-receiver_{context}"]
+=== Kubelet Stats Receiver
+
+:FeatureName: The Kubelet Stats Receiver
+include::snippets/technology-preview.adoc[]
```
Comment (mletalie) [inline modules/otel-collector-components.adoc:89]: Not sure if it is outside the scope of this ticket, but I noticed the subtopics below this one have a link to the Technology Preview blurb  (as opposed to the snippet). I think it isn't consistent. Not sure if they need to be changed or the snippet should be a link to match those below?

**Review by mletalie — COMMENTED** [review]
/lgtm
/remove-label peer-review-needed
/remove-label peer-review-in-progress
/label peer-review-done

File: modules/otel-collector-components.adoc [hunk 8]
```diff
@@ -82,6 +82,55 @@ The Jaeger receiver ingests traces in the Jaeger formats.
...
+        fieldPath: spec.nodeName
+----
+<1> Sets the `K8S_NODE_NAME` to authenticate to the API.
+
+The Kubelet Stats Receiver requires additional permissions for the service account used for running the OpenTelemetry Collector.
+
+.Permissions required by the service account
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
```
Comment (IshwarKanse) [inline modules/otel-collector-components.adoc:120]: The nodes/stats need to have get,watch, list verbs. 

```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: otel-collector
rules:
  - apiGroups: ['']
    resources: ['nodes/stats']
    verbs: ['get', 'watch', 'list']
  - apiGroups: [""]
    resources: ["nodes/proxy"]
    verbs: ["get"]
```

Comment (IshwarKanse) [reply modules/otel-collector-components.adoc:120]: Apart from this rest LGTM.

**Review by IshwarKanse — APPROVED** [review]

File: modules/otel-collector-components.adoc (hunk 8 above)
Comment (max-cx) [reply modules/otel-collector-components.adoc:122]: Thank you. Updated.

**Comment by max-cx** [comment]
> Hello @max-cx, Please remember to squash down to one commit before requesting merge. Thanks!

@mletalie, thank you so much.

**Comment by max-cx** [comment]
Closing this PR because I cherry-picked its commit into https://github.com/openshift/openshift-docs/pull/76340.

**Pull request merged**